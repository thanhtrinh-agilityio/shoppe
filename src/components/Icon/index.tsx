import React, { memo } from 'react'
import Image, { ImageLoader } from 'next/image'
import { Box, BoxProps } from '@chakra-ui/react'

// Services
import myImageLoader from '@services/imageLoader'

interface IconProps extends BoxProps {
  srcIcon: string
  alt: string
}

const Icon = ({ srcIcon, alt, ...rest }: IconProps) => {
  return (
    <Box position="relative" display="flex" alignItems="flex-end" {...rest}>
      <Image
        src={srcIcon}
        alt={alt}
        data-testid="image"
        fill
        loader={myImageLoader as ImageLoader}
      />
    </Box>
  )
}

export default memo(Icon)
