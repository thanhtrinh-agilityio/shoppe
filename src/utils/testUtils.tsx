import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

// Contexts
import { CartProvider } from '@contexts/CartProvider'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>
      <CartProvider>{component}</CartProvider>
    </ChakraProvider>,
  )
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { customRender as render }
