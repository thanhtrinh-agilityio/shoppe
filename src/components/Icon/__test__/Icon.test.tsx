import React from 'react'

// component
import Icon from '@components/Icon'

// Utils
import { render, screen } from '@utils/testUtils'

describe('Icon render', () => {
  const props = {
    alt: 'facebook',
    srcIcon: '/images/facebook.svg',
  }

  it('Should show match Icon DOM Snapshot', () => {
    const { container } = render(<Icon {...props} />)
    expect(container).toMatchSnapshot()
  })

  it('Should image has the correct src', () => {
    render(<Icon {...props} />)

    const image = screen.getByTestId('image')

    expect(image.getAttribute('src')).toEqual(`${props.srcIcon}?w=3840&q=75`)
  })
})
