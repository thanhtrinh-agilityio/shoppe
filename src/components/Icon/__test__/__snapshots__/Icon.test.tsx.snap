// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon render Should show match Icon DOM Snapshot 1`] = `
<div>
  <div
    class="css-fm4v0q"
  >
    <img
      alt="facebook"
      data-nimg="fill"
      data-testid="image"
      decoding="async"
      loading="lazy"
      sizes="100vw"
      src="/images/facebook.svg?w=3840&q=75"
      srcset="/images/facebook.svg?w=640&q=75 640w, /images/facebook.svg?w=750&q=75 750w, /images/facebook.svg?w=828&q=75 828w, /images/facebook.svg?w=1080&q=75 1080w, /images/facebook.svg?w=1200&q=75 1200w, /images/facebook.svg?w=1920&q=75 1920w, /images/facebook.svg?w=2048&q=75 2048w, /images/facebook.svg?w=3840&q=75 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
